app/gui/docs/contributing/style-guide.md

```diff
@@ -2,46 +2,52 @@
 layout: style-guide title: Rust Style Guide category: style-guide tags: [style-guide,contributing]
 ---
 
-# Rust Style Guide
-
-We are using `rustfmt` for the basic formatting of our rust code, which is also
-checked on CI. We have some additional guidelines for imports, naming, sections
-within files and naming which are documented here.
-
-## Styling rules
-
-### Imports
-
-Imports should be divided into 4 groups separated by blank lines. Items in the
-groups should be sorted alphabetically.
-
-```rust
-// Group 1: sub-module definitions.
-// Group 2: prelude-like imports.
-// Group 3: local-crate imports.
-// Group 4: external imports.
-```
-
-For example:
-
-```rust
-pub mod display_object;
-
-use crate::prelude::*;
-
-use crate::closure;
-use crate::data::opt_vec::OptVec;
-use crate::data::dirty;
-use crate::system::web::group;
-
-use nalgebra::Matrix4;
-use nalgebra::Vector3;
-```
-
-### Sections
+# Rust style guide.
+
+All of the codebase should be formatted by the `rustfmt`. However, the code style is way more than just formatting. In many cases formatting can be automated. According to rustfmt docs: “formatting code is a mostly mechanical task which takes both time and mental effort. By using an automatic formatting tool, a programmer is relieved of this task and can concentrate on more important things.”. While in many cases it is true, if the code author does not take extra effort to make his code pretty by refactoring long lines to variables, moving code to specific modules, or sections, the formatting tool will result in code that is hard to read and hard to write. Thus, it is important to take time to write code in such way that we can be proud of its quality. The following document provides you with a detailed guide regarding the code quality we are looking for.
+
+## Submodules and imports.
+
+- **Design your files to be imported as module.**  
+  Design names of your libraries, structs, and functions to be imported as modules. For example, prefer an import `use graph;` and it's usage `graph::Node::new()` over `use graph::new_node`. This design minimizes the amount of imports and allows related modules to import shorter names to the scope.
+
+- **Use one import per line.**  
+  Do not use the `use crate::mod_name::{name1, name2, name3}` syntax. For each entity use a separate `use` statement.
+  
+- **Don't use relative imports.**  
+  Do not use `super::` nor `self::` imports. Use absolute imports or imports from local submodules only. 
+
+- **Group related submodule and import statements**  
+  Submodule and import statements should be divided into several groups separated by blank lines. Items in the groups should be sorted alphabetically (do not use group comments in your code):
```

Comment:
This is not what I've been already doing nor what I've been requesting in my reviews, so I think it actually warrants some discussion.
Also, this should be unified with https://github.com/enso-org/ide/blob/develop/docs/contributing/style-guide.md
Ideally, both style guides should be unified, as we have a single repository now.
Merging as-is would create a conflict between these documents.

Reply:
> Also, this should be unified with https://github.com/enso-org/ide/blob/develop/docs/contributing/style-guide.md
Ideally, both style guides should be unified, as we have a single repository now.
Merging as-is would create a conflict between these documents.

But the enso-org/ide is archived, right? That strongly suggests, that the guidelines there should not be used anymore.

Reply:
Ouch, wrong link.
The one I've meant is https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md

Issue is that we have two Rust style guide documents.

Reply:
The idea is to remove the second style guide ASAP and make this the only one.